packages/ember/tests/routing/query_params_test.js

```diff
@@ -1963,7 +1963,66 @@ if (isEnabled('ember-routing-route-configured-query-params')) {
     equal(get(controller, 'bar'), 'rab');
     equal(get(controller, 'foo'), '456');
   });
+
+  QUnit.test('Calling transitionTo does not loose query params already on the activeTransition', function() {
```

Comment:
This would be amazing if this issue was finally fixed, nice work! Also, tiny spelling correction here - I think you meant `lose`, not `loose`.